### PR TITLE
make release.sh idiot (even Kirk) proof

### PR DESCRIPTION
**Pull request by Kirkkt**
@dmitrig01 

Effectively an NOP.

Update `release.sh` according to this work flow:

- update **CHANGELOG.md** on github, no PR needed, just edit directly.
- `git pull origin master` in local master branch
- run release.sh, keep pushing y vigorously
- update documentation repo
    - <https://github.com/BranchMetrics/documentation/edit/master/ingredients/web_sdk/_initialization.md>
    - <https://github.com/BranchMetrics/documentation/edit/master/ingredients/web_sdk/send_sms_example.md>
- post-release check
    - <https://cdn.branch.io/example.html>
    - <https://cdn.branch.io/branch-v.version.number.min.js>
    - <https://github.com/BranchMetrics/Smart-App-Banner-Deep-Linking-Web-SDK/blob/v.version.number/CHANGELOG.md>

One last thing, also stop ignoring -e temp files since... why not :D

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -31,7 +31,6 @@ npm-debug.log
 tmp
 .map
 Thumbs.db
-*.*-e
```
Comment (dmitrig01) [inline .gitignore]: we probalby want to keep this?

File: release.sh [hunk 2]
```diff
@@ -6,6 +6,12 @@ VERSION_NO_V=$(echo $1 | tr -d "\nv")
 VERSION="v"$VERSION_NO_V
 DATE=$(date "+%Y-%m-%d")
 
+echo "Sanity checks"
+
+read -p "Are you on the master branch?" -n 1 -r
```
Comment (dmitrig01) [inline release.sh]: let's make this check automated?

File: release.sh [hunk 3]
```diff
@@ -6,6 +6,12 @@ VERSION_NO_V=$(echo $1 | tr -d "\nv")
...
+read -p "Have you amended CHANGELOG.md and git push it?" -n 1 -r
```
Comment (dmitrig01) [inline release.sh]: let's make use a git command to make sure there are no changes in the working tree or index (i.e. nothing uncommitted)

File: .gitignore (hunk 1 above)
Comment (Kirkkt) [inline .gitignore]: Would you kindly enlighten? Why do we want to keep this _across commits_?

I thought we want to clean up after running release.sh. I.e. why do we want to check -e files into github in the first place?

Comment (dmitrig01) [inline .gitignore]: we want to keep the line in the gitignore, so you can't commit it into git even if it mistakenly makes it out of release.sh

**Comment by dmitrig01** [comment]
:+1:

**Comment by Kirkkt** [comment]
thanks! almost forgot this :)

**Pull request merged**